src/frontend/src/lib/stores/auth.store.ts

```diff
@@ -85,6 +89,14 @@ const initAuthStore = (): AuthStore => {
 				...state,
 				identity: null
 			}));
+		},
+
+		setForTesting: (identity) => {
```

Comment:
WOWOWOW! Eheh! Now I understand why you are not a fan... hardcode a test-specific function

I have to say, that it is a smart pattern, but not ideal (not a blocker, just an opinion). Maybe add a `TODO` for posterity to think of something  else?

Reply:
In the NNS dapp, we used various solutions, including mocking the entire sign-in process, the `AuthClient` class, the store, etc. Ultimately, this was the final pattern we adopted.

Have a look to the test suite there and let me know if one or the other pattern better suits you. For sure happy to user another one.

Reply:
Not really any better idea! I would just put something as "documentation", maybe instead of `TODO` tag, use `HACK` tag?

Reply:
@AntonioVentilii-DFINITY can you please review this commit https://github.com/dfinity/oisy-wallet/pull/2677/commits/ee03926840bdedc1be92e917c665e5ff6b6f874b or the PR again?

- I added the documenation with warning
- I created a test (needed a second commit because I forgot to update a title https://github.com/dfinity/oisy-wallet/pull/2677/commits/cd7333e3b113669985ec8530a56286dc6f4d66a3)
- And actually discovered that the TEST constant was not correctly set

Let me know what you think.